src/shared/containers/ResourceViewContainer.tsx

```diff
@@ -608,14 +606,64 @@ const ResourceViewContainer: React.FunctionComponent<{
     { key: 'jira', name: 'jira', pluginComponent: jiraPlugin },
     { key: 'analysis', name: 'Analysis', pluginComponent: analysisPlugin },
   ];
-  React.useEffect(() => {
+
+  useEffect(() => {
     return () => {
       dispatch({
         type: UISettingsActionTypes.UPDATE_CURRENT_RESOURCE_VIEW,
         payload: null,
       });
     };
   }, []);
+
+  function constructUnDeprecateUrl(
+    apiEndpoint: string,
+    resource: Resource,
+    latestResource: Resource,
+    orgLabel: string,
+    projectLabel: string
+  ) {
+    return `${apiEndpoint}/${
+      resource!['@type'] === 'File' ? 'files' : 'resources'
+    }/${orgLabel}/${projectLabel}/${
+      resource!['@type'] === 'File' ? '' : '_/'
+    }${encodeURIComponent(resource!['@id'])}/undeprecate?rev=${
+      latestResource!._rev
+    }`;
+  }
+
+  const { mutate: unDeprecateResource } = useMutation({
+    mutationFn: async () => {
+      try {
+        await nexus.httpPut({
+          path: constructUnDeprecateUrl(
+            apiEndpoint,
+            resource!,
+            latestResource!,
+            orgLabel,
+            projectLabel
+          ),
+        });
+
+        setLatestResource({
+          ...latestResource!,
+          _rev: latestResource!._rev + 1,
+          _deprecated: false,
+        });
```

Comment:
no need to set the state since there is a redirection just after

Reply:
Good point. I double-checked and indeed it doesn't make a difference other than not letting React decide to update the state. However, see my reply on your other comment.